Realm/RLMObjectBase.mm

```diff
@@ -750,6 +751,9 @@ uint64_t RLMObjectBaseGetCombineId(__unsafe_unretained RLMObjectBase *const obj)
 }
 
 @implementation RealmSwiftObject
++ (RLMObjectSchema *)sharedSchema {
```

Comment:
This is not a correct implementation for RealmSwiftObject. DynamicObject is what needs this, not Object.

Reply:
So, another weird issue I have found is that if the 'dynamic object from mixed' code is run (`RLMObjectFromObjLink`) before the test `testInvalidObjects()` objects with subclasses will have duplicated properties (only properties from the base class are showing up). Preventing `maybeInitObjectSchemaForUnmanaged` being called from `RLMObjectFromObjLink` solves the issue but its probably pointing to something more sinister that I have yet to find the root cause for.

Reply:
Add this solved the failing test: https://github.com/realm/realm-cocoa/blob/53244c2cc555c5c99a304cb0c796f49076407757/Realm/RLMObjectBase.mm#L759

So now when constructing the dynamic mixed object, the accessor class will be RealmDynamicSwiftObject